resources/js/Components/CharacterDetail/CharacterDetailBox.tsx

```diff
@@ -3,10 +3,12 @@ import DownloadIcon from '@mui/icons-material/Download'
 import IosShareIcon from '@mui/icons-material/IosShare'
 import {Box, Button, Grid, Stack, Typography} from '@mui/material'
 import {Link} from 'Components'
-import React from 'react'
+import React, {useContext} from 'react'
```

Comment:
as much as I like the use of a new hook, this is a decent approach but there are other better ways of doing this.
For example, we could use a state management library such as `redux`
Or since the backend is passing down `user` as a global object, you can just use a `Inertia` hook `usePage` to get the user object or even pass down from the parent component since it will exist there

Reply:
https://inertiajs.com/shared-data